Loop/Managers/LoopDataManager.swift

```diff
@@ -1078,6 +1092,98 @@ extension LoopDataManager {
         return prediction
     }
 
+    fileprivate func predictGlucoseFromManualGlucose(
+        _ glucose: NewGlucoseSample,
+        potentialBolus: DoseEntry?,
+        potentialCarbEntry: NewCarbEntry?,
+        replacingCarbEntry replacedCarbEntry: StoredCarbEntry?,
+        includingPendingInsulin: Bool
+    ) throws -> [PredictedGlucoseValue] {
+        let retrospectiveStart = glucose.date.addingTimeInterval(-retrospectiveCorrection.retrospectionInterval)
+        let earliestEffectDate = Date(timeIntervalSinceNow: .hours(-24))
+        let nextEffectDate = insulinCounteractionEffects.last?.endDate ?? earliestEffectDate
+
+        let updateGroup = DispatchGroup()
+        let effectCalculationError = Locked<Error?>(nil)
+
+        var insulinEffect: [GlucoseEffect]?
```

Comment:
Glucose effects aren't computed when glucose data is stale, since no prediction is vended. To offer a prediction from manual glucose, these effects are computed locally on demand.